File: DraftPuck.Api/Services/NhlService.cs
using Draftpuck.Nhl.Services.Interfaces;

namespace DraftPuck.Api.Services
{
    public class NhlService : INhlService
    {
        private readonly INhlApiService _nhlApi;
        private readonly IMapper _mapper;

        public NhlService(INhlApiService nhlApi, IMapper mapper)
        {
            _nhlApi = nhlApi;
            _mapper = mapper;
        }

        public async Task<Schedule> GetScheduleByDateAsync(DateTime date)
        {
            var schedule = await _nhlApi.GetScheduleAsync(date);
            return _mapper.Map<Schedule>(schedule);
        }

        public async Task<Player> GetPlayerAsync(int playerId)
        {
            var player = await _nhlApi.GetPlayerAsync(playerId);
            return _mapper.Map<Player>(player);
        }

        public async Task<Game> GetGameAsync(int gameId)
        {
            var game = await _nhlApi.GetFullGameAsync(gameId);
            return _mapper.Map<Game>(game);
        }
    }
}

File: DraftPuck.Api/ClientApp/src/services/NhlService.ts
import HttpService, { type IHttpService } from '@/services/HttpService'

const controller = 'nhl'

class NhlApiService {
  private readonly _http: IHttpService

  constructor() {
    this._http = new HttpService(controller)
  }

  public async getSchedule(date: string): Promise<Schedule> {
      return this._http.get(`schedule/${date}`)
  }
  
  public async getGame(gameId: number): Promise<Game> {
    return this._http.get(`game/${gameId}`)
  }
}

export default new NhlApiService()
